Sources/NIO/EventLoopFuture.swift

```diff
@@ -667,9 +682,16 @@ extension EventLoopFuture {
     /// - returns: A future that will receive the recovered value.
     @inlinable
     public func recover(file: StaticString = #file, line: UInt = #line, _ callback: @escaping (Error) -> Value) -> EventLoopFuture<Value> {
-        return self.flatMapError(file: file, line: line) {
-            return EventLoopFuture<Value>(eventLoop: self.eventLoop, value: callback($0), file: file, line: line)
+        let next = EventLoopPromise<Value>(eventLoop: eventLoop, file: file, line: line)
+        self._whenComplete {
+            switch self._value! {
```

Comment:
Similar note to above, can we use `self._value!.mapError` here?

Reply:
`mapError` sadly doesn't work here. Because it maps `Error` to a new `Error`. But in this case we want to map `Error` -> `Value`.

Reply:
Oh yes, my mistake!